packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/knowledge/KnowledgeController.java

```diff
@@ -106,6 +122,38 @@ ResponseEntity<ExtractionResponse> postCodeToAMR(
 						.getBody());
 	}
 
+		// Create a model from code blocks
+	@PostMapping(value = "/code-blocks-to-model", consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
+	@Secured(Roles.USER)
+	public ResponseEntity<ExtractionResponse> codeBlocksToModel(@RequestPart Code code, @RequestPart("file") MultipartFile input) throws IOException {
```

Comment:
You're also able to add swagger docs via annotations here now. Something like...
```
@Operation(summary = "Create a model from code blocks")
	@ApiResponses(value = {
			@ApiResponse(responseCode = "200", description = "BLAH BLAH", content = @Content(mediaType = "application/json", schema = @io.swagger.v3.oas.annotations.media.Schema(implementation = ExtractionResponse.class))),
			@ApiResponse(responseCode = "500", description = "BLAH BLAH", content = @Content)
	})
```

Reply:
good to know, added